iceoryx_dds/include/iceoryx_dds/dds/data_reader.hpp

```diff
@@ -60,13 +71,16 @@ class DataReader
 
     ///
     /// @brief take Take the next available sample from the DDS data space.
-    /// @param buffer Receive buffer in which sample will be stored.
-    /// @param bufferSize Size of the provided buffer.
+    /// @param datagramHeader with size information
+    /// @param userHeaderBuffer buffer for the user-header
+    /// @param userPayloadBuffer buffer for the user-payload
     /// @return Error if unsuccessful.
     ///
-    virtual iox::cxx::expected<DataReaderError> takeNext(uint8_t* const buffer, const uint64_t& bufferSize) = 0;
+    virtual iox::cxx::expected<DataReaderError>
+    takeNext(const IoxChunkDatagramHeader datagramHeader, uint8_t* userHeaderBuffer, uint8_t* userPayloadBuffer) = 0;
 
 
+    /// @todo iox-#14 this needs either to be removed or also ported to IoxChunkDatagramHeader
```

Comment:
@ithier can the `DataReader::take` method be removed?

Reply:
The idea behind this method was to transfer multiple dds samples to iceoryx in one go. 
I don't think it actually does that at the moment and probably needs to be improved. 
If you don't foresee this use-case being useful though, then it could be removed.

Reply:
well, I didn't port it to the IoxChunkDatagramHeader because I didn't knew the use case since it was only used in the tests. If you don't mind, I would just remove it for the moment